### PR TITLE
20 Load events to calendar for a specific worker

### LINKED ISSUE #20: Load events to calendar for a specific worker
Get worker's events for a specific week
Try not to use GetAllForWorker, but GetFromXToYForWorker() which means that load only 1 week's event at first, and then based on which week is selected load those events.

EST 5

### PR DESCRIPTION
**Pull request by akosgdcsi**
getting things done

### CODE REVIEW

**Review by luxz97 — APPROVED** [review]

File: wman-backend/Wman.Logic/Classes/CalendarEventLogic.cs [hunk 1]
```diff
@@ -138,5 +142,147 @@ public async Task<List<WorkEventForWorkCardDTO>> GetWeekEvents(DateTime firstDay
...
+                    firstDayOfTheWeek = time - TimeSpan.FromDays(5);
+                    lastDayOfTheWeek = time + TimeSpan.FromDays(1);
+                    break;
+                default:
+                    break;
+            }
+
+
+            var events = (from x in user.WorkEvents
+                          where x.EstimatedStartDate.DayOfYear >= firstDayOfTheWeek.DayOfYear && x.EstimatedStartDate.DayOfYear <= lastDayOfTheWeek.DayOfYear
```
Comment (pop-simon) [inline wman-backend/Wman.Logic/Classes/CalendarEventLogic.cs]: The constraint is almost good but it shall contain something like this: `x.EstimatedStartDate.Year == DateTime.Today.Year`
**This error comes up in every method in this class so please fix all of them.**

**Review by pop-simon — CHANGES_REQUESTED** [review]
**There are several problems with the implementation.**
- You do not follow the DRY principles. The four method you just implemented are already exist. The only difference is that the existing methods do not use filtering by `username`.
- This feature enables a worker to get their `WorkEvent`s. It is not allowed to get another worker's `WorkEvent`s for the currently logged in worker.
---
**There are two ways to implement this feature in my opinion.**
- You could extend the already existing methods with user checking. 
For example the method `public async Task<List<WorkEventForWorkCardDTO>> GetCurrentDayEvents()` can be modified like this:
```csharp
public async Task<List<WorkEventForWorkCardDTO>> GetCurrentDayEvents(string username)
        {
            var user = await userManager.Users
                .Where(x => x.UserName == username)
                .SingleOrDefaultAsync();
            List<WorkEvent> events = await (from x in workEventRepo.GetAll()
                                            where x.EstimatedStartDate.DayOfYear == DateTime.UtcNow.DayOfYear && x.EstimatedStartDate.Year == DateTime.Today.Year
                                            select x).ToListAsync();

            if (await userManager.IsInRoleAsync(user, "worker"))
            {
                events = events.Where(x => x.AssignedUsers.Any(x => x.UserName == username)).ToList();
            }

            return mapper.Map<List<WorkEventForWorkCardDTO>>(events);
        }
```
In this case you have to modify the related **interface** and **controller** as well. Furthermore in `AuthLogic.cs` the `claims` starting at line 136 have to look like this:
```csharp
var claims = new List<Claim>
                {
                  new Claim(JwtRegisteredClaimNames.Sub, user.UserName),
                  new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
                  new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
                  new Claim(ClaimTypes.Name, user.UserName)
                };
```
- The other solution is to keep the new methods but those should use the existing ones and then filter the results.

In both solutions you should get the `username` from the `HttpContext` as Zoli did in the previous issue's implementation #56 in `WorkerController.cs`.

I am sure that there are other ways to fix this but only these two solutions came to my mind.

**Review by tsaskater — CHANGES_REQUESTED** [review]
Works good. Very clever solution.
I need some refactoring. Please add a field as phoneNumber for the workers because it's missing at the moment.

**Comment by akosgdcsi** [comment]
Time to ruin somebodys Xmas

**Comment by akosgdcsi** [comment]
kép

**Comment by tsaskater** [comment]
> Time to ruin somebodys Xmas

image

**Comment by akosgdcsi** [comment]
image

Me vibing right now

**Comment by rolandbogdan** [comment]
orban-viktor-1
Me seeing that I got added as a reviewer to another WMAN PR...

**Review by tsaskater — APPROVED** [review]
image
🎄🤶🏿

**Review by pop-simon — APPROVED** [review]

**Pull request merged**